modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIEndpointRelevantObjectEntryModelListener.java

```diff
@@ -236,14 +269,18 @@ private void _validateAPISchema(long apiApplicationId, long apiSchemaId)
 	}
 
 	private void _validateSingleElementPath(
-			ObjectEntry objectEntry, String pathParameter, String pathString)
+			ObjectEntry objectEntry, String pathParameter, String pathString,
+			long responseAPISchemaId)
 		throws Exception {
 
-		if (pathParameter == null) {
+		if (!Validator.isBlank(pathParameter) && (responseAPISchemaId != 0) &&
+			!_isValidPathParameter(
+				objectEntry.getCompanyId(), pathParameter,
+				responseAPISchemaId)) {
+
 			throw new ObjectEntryValuesException.InvalidObjectField(
-				null,
-				"Path parameter cannot be null in a single element endpoint",
-				"path-parameter-cannot-be-null-in-a-single-element-endpoint");
+				null, "Path parameter must be a valid field name",
```

Comment:
I think the description is not very clear for a user. Which would be a valid field name? Maybe we could discuss internally with the team.

What about "Path parameter must be an id, external reference code or unique field" ?

Reply:
I'll ask it in the internal channel of Headless

Reply:
As PM members response, the valid one is yours. So I'll change it